### PR TITLE
Enable Public Key for new user accounts

### PR DESCRIPTION
**Pull request by dorileo**
Fixes Issue: #74 

Changes proposed in this pull request:
- Enable Public Key for new user accounts
- Allow users to be created without password

### CODE REVIEW

File: user/user.go [hunk 1]
```diff
@@ -186,17 +188,56 @@ func (u *User) apply(rootDir string) error {
...
-	if err := cmd.PipeRunAndLog(pwd, args...); err != nil {
-		return errors.Wrap(err)
+	return nil
+}
+
+func writeSSHKey(rootDir string, u *User) error {
+	dpath := filepath.Join(rootDir, "home", u.Login, ".ssh")
+	fpath := filepath.Join(dpath, "authorized_keys")
+
+	if err := utils.MkdirAll(dpath, 0755); err != nil {
```
Comment (mdhorn) [inline user/user.go]: These permissions are too permissive
```suggestion
	if err := utils.MkdirAll(dpath, 0700); err != nil {
```

File: user/user.go [hunk 2]
```diff
@@ -186,17 +188,56 @@ func (u *User) apply(rootDir string) error {
...
+		return err
+	}
+
+	f, err := os.OpenFile(fpath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0644)
```
Comment (mdhorn) [inline user/user.go]: These permissions are too permissive
```suggestion
	f, err := os.OpenFile(fpath, os.O_RDWR|os.O_CREATE|os.O_TRUNC, 0600)
```

**Comment by mdhorn** [comment]
@dorileo After the permission changes, can you run a mass install with a real ssh key and make sure the created user can login with ssh.

**Comment by dorileo** [comment]
@mdhorn updated the permission also change to adjust the file's owner. And yes, it's tested.

Thanks...

**Pull request merged**